draft-colitti-ipsecme-esp-ping.mkd

```diff
@@ -74,8 +75,74 @@ An IPv6 node that desires to determine whether the path to a particular destinat
+Should the destination support ESP and intend to communicate this capability to the potential IPsec peer, it SHOULD respond with an ESP Echo Reply packet.
+
+The sender MAY send ESP Echo packets with zero data. 
+When responding to an ESP Echo packet, the node SHOULD generate a packet devoid of any data.
+If the ESP Echo Request packet contain non-zero data, the resulting ESP Echo Reply packet size MUST NOT exceed the size of the ESP Echo Request packet received.
+
+# Discovering ESP Echo Support
+
+If no response is received to an ESP Echo Request packet, it can be caused by one of the following:
+
+* the peer doesn't support ESP Echo protocol.
+* there is no end-to-end ESP connectivity.
+
+Without some prior knowledge about ESP Echo support by the remote side, the sender can not distibguish those two scenarios.
+Therefore the sender SHOULD NOT treat lack of response as an indicator of end-to-end connectivity issues until an explicit confirmation of ESP Echo support by the peer is received. 
+The IPSec peer SHOULD announce ESP Echo support using IKE Notify Payload (more text to be added heve in the next version).
+The sender MAY use any other means of obtaining the information about  ESP Echo support, such as an explicit out-of-band configuration (for example, a VPN client might be configured to always use ESP Echo when communicating to the given VPN server).
+
+# Updates to RFC4303 
+
+Section 2.6 of {{!RFC4303}} discusses "dummy" ESP packets, which are distinguishable by the Next Header value set to 59.
+As per {{!RFC4303}} a receiver MUST be prepared to silently discard "dummy" packets. This document updates Section 2.6 of {{!RFC4303}} to allow packets with the Next Header value of 59 to be processed, if SPI is set to [ESP-ECHO-REQUEST] or [ESP-ECHO-REPLY].
+
+OLD TEXT:
+
+A transmitter MUST be capable of generating dummy packets marked with this value in the next protocol field, and a receiver MUST be prepared to discard such packets, without indicating an error.
+
+NEW TEXT:
+
+A transmitter MUST be capable of generating dummy packets marked with this value in the next protocol field, and a receiver MUST be prepared to discard such packets, without indicating an error.
+A transmitter MUST NOT use the reserved SPI values [ESP-ECHO-REQUEST] or [ESP-ECHO-REPLY] for dummy packets. A receiver SHOULD NOT discard packets with  the Next Header value set of 59, if those packets use the reserved SPI values.
+Packets with the reserved SPI values [ESP-ECHO-REQUEST] or [ESP-ECHO-REPLY] and the Next Header value set of 59 SHOULD be processed by the receiver as described in draft-colitti-ipsecme-esp-ping.
+
+
 # Security Considerations
 
+To prevent a downgrade attack, the IPSec-capable node MUST NOT fall back to unencrypted mode of communication in case of ESP Echo failure.
+The node MAY switch to another path (e.g. via another interface) or another protocol (e.g. IPv4).
```

Comment:
```suggestion
To prevent a downgrade attack, the IPsec-capable node MUST NOT fall back to unencrypted mode of communication in case of ESP Echo failure.
The node SHOULD switch to another method: another interface, a different encapsulation, or a different protocol (e.g., IPv4).
```

Reply:
If we are saying SHOULD, then I think we need to phrase it as 'if such methods are available'. Because IPSec over IPv6 can be the only available way of communication. Also, we might want to say "...unless explicitly configured otherwise", to allow the explicitly configured policy to override the RFC text.
What do you think?

Reply:
+1 to MAY. For example: even if the ping fails, the client might want to try IPv6 ESP anyway, because it might work (e.g., if the server does not support ESP ping, or if the ping was dropped).

MAY still allows a client that knows for sure that the server supports ESP ping to fall back to other options if it gets no ping reply. We could say something like:

====
The node MAY switch to another path (e.g. via another interface) or another protocol (e.g. IPv4). If the client is certain that the server supports ESP ping, it SHOULD switch to another path or another protocol, because the lack of a ping response means that IPv6 ESP is not likely to work on the current path.
====

but I think a simple MAY is fine as well (without this extra text I'm proposing).

Reply:
I think that the key thing is that lack of an ESP Ping reply SHOULD NOT result in a cleartext mandate.

Reply:
>I think that the key thing is that lack of an ESP Ping reply SHOULD NOT result in a cleartext mandate.

I believe we are in a violent agreement about that, and the current next actually says MUST NOT even.